### PR TITLE
docs: adding documentation for regressors

### PR DESCRIPTION
**Pull request by YibinSun**
-- Add documenting for regressors:
```FIMTDD```
```ORTO```
```KNNRegressor```
```Passive Aggressive Regressor```
```SGD Regressor```
```SOKNL```
```SOKNLBT```
-- Update `__init__.py` accordingly

### CODE REVIEW

File: src/capymoa/regressor/_soknl.py [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
-from capymoa.splitcriteria import SplitCriterion, _split_criterion_to_cli_str
-from capymoa.stream._stream import Schema
-from moa.classifiers.meta import SelfOptimisingKNearestLeaves as _MOA_SOKNL
-from ._soknl_base_tree import SOKNLBT
-
 
 class SOKNL(MOARegressor):
+    """Self-Optimising K-Nearest Leaves (SOKNL) Implementation.
+
+    SOKNL is an extension to AdaptiveRandomForestRegressor.
```
Comment (hmgomes) [inline src/capymoa/regressor/_soknl.py:14]: SOKNL extends the AdaptiveRandomForestRegressor by limiting the number of base trees (or tree leaves) involved in predicting a given instance. This approach overrides the aggregation strategy used for voting, leading to more accurate predictions in general.

File: src/capymoa/regressor/_soknl.py [hunk 2]
```diff
@@ -1,17 +1,40 @@
...
+    SOKNL stores abstract information for all instance seen by the leaves, use them to calculate the distance
```
Comment (hmgomes) [inline src/capymoa/regressor/_soknl.py:15]: Improve this part. What is the abstract information? What is the certain instance? You may refer to it as the instance whose target value is being predicted (or find a more concise way to refer to it :) ). Also, you are not explaining the Self-Optimising mechanism in here, and you should.

File: src/capymoa/regressor/_soknl.py [hunk 3]
```diff
@@ -1,17 +1,40 @@
...
+    from the leaf to a certain instance. Then final predictions are yielded by the closest k leaves in the forest.
+
```
Comment (hmgomes) [inline src/capymoa/regressor/_soknl.py:17]: Before the reference add a "See also" for ARFReg and MOARegressor. You can refer to other classes using the syntax below (See _adaptive_random_forest.py for reference on the syntax as well)

See also :py:class:\`capymoa.regressor.AdaptiveRandomForestRegressor\`
See :py:class:\`capymoa.base.MOARegressor\` for train and predict.

File: src/capymoa/regressor/_soknl_base_tree.py [hunk 4]
```diff
@@ -11,7 +11,29 @@
 
 
 class SOKNLBT(MOARegressor):
-    """The base tree for Self-Optimising K Nearest Leaves as distribed by Sun. at el."""
+    """The base tree for Self-Optimising K Nearest Leaves as distribed by Sun. at el.
+
+    SOKNLBT is a modification to the FIMT-DD that allows FIMT-DD to store necessary information.
+
+    See in reference:
```
Comment (hmgomes) [inline src/capymoa/regressor/_soknl_base_tree.py:18]: Just Reference, not "See in reference"

File: src/capymoa/regressor/_soknl_base_tree.py [hunk 5]
```diff
@@ -11,7 +11,29 @@
 
 
 class SOKNLBT(MOARegressor):
-    """The base tree for Self-Optimising K Nearest Leaves as distribed by Sun. at el."""
+    """The base tree for Self-Optimising K Nearest Leaves as distribed by Sun. at el.
+
+    SOKNLBT is a modification to the FIMT-DD that allows FIMT-DD to store necessary information.
```
Comment (hmgomes) [inline src/capymoa/regressor/_soknl_base_tree.py:16]: SOKNLBT modifies the FIMT-DD algorithm to store information at the leaves that allows SOKNL to calculate the distance between a given instance and a leaf. 

See also... (add link to SOKNL) and (add link to FIMTDD)

**Review by hmgomes — CHANGES_REQUESTED** [review]
Please see the request changes. Most of them are related to the content of the documentation for SOKNL, we need to make it more explicit what the algorithm does.

**Comment by YibinSun** [comment]
@hmgomes, amendments have been implemented, please take a look.

**Comment by hmgomes** [comment]
thanks @YibinSun , it is looking much better now

**Review by hmgomes — APPROVED** [review]

**Pull request merged**